mlrun/datastore/dbfs_store.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import fsspec
+
+import mlrun.errors
+from mlrun.datastore.helpers import ONE_MB
+
+from .base import DataStore, FileStats
+
+# dbfs objects will be represented with the following URL: dbfs://<path>
+
+
+class DBFSStore(DataStore):
+    def __init__(self, parent, schema, name, endpoint="", secrets: dict = None):
+        super().__init__(parent, name, schema, endpoint, secrets=secrets)
+        self.get_filesystem(silent=False)
+
+    def get_filesystem(self, silent=True):
+        """return fsspec file system object, if supported"""
+        if not self._filesystem:
+            self._filesystem = fsspec.filesystem("dbfs", **self.get_storage_options())
+        return self._filesystem
+
+    def get_storage_options(self):
+        return dict(
+            token=self._get_secret_or_env("DATABRICKS_TOKEN"), instance=self.endpoint
+        )
+
+    def _prepare_path_and_verify_filesystem(self, key: str):
```

Comment:
So the name makes sense in the GCS implementation because it actually returns a different path. This one only validates (so might as well return nothing at all).

Another thing, I know this is copied from the GCS implementation, but either it's a path or a key. Using the two interchangeably is a little confusing.